CondFormats/CTPPSReadoutObjects/src/LHCOpticalFunctionsSet.cc

```diff
@@ -0,0 +1,137 @@
+// Original Author:  Jan Kašpar
+
+#include "CondFormats/CTPPSReadoutObjects/interface/LHCOpticalFunctionsSet.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "TFile.h"
+#include "TGraph.h"
+
+//----------------------------------------------------------------------------------------------------
+
+LHCOpticalFunctionsSet::LHCOpticalFunctionsSet(const std::string &fileName, const std::string &directoryName, double z) :
```

Comment:
It can, but I find it more logical that it is a constructor which builds an object from a data file.

Reply:
the driving argument is for a simplicity in software (library) dependencies.
Somewhat similarly, we do not allow regular data formats to depend on edm::Event (even though it may feel trivial in some cases to allow bootstrap).
The consequence is that with any non-trivial complexity on the side of making an object of a given type and that of using the object for other computations, the object's type implementation should be as simple as possible.